packages/quantic/force-app/examples/main/lwc/exampleRelevantGenerativeAnswering/exampleRelevantGenerativeAnswering.js

```diff
@@ -0,0 +1,82 @@
+import {LightningElement, api} from 'lwc';
+// @ts-ignore
+import caseTemplate from './resultTemplates/caseResultTemplate.html';
+// @ts-ignore
+import chatterTemplate from './resultTemplates/chatterResultTemplate.html';
+// @ts-ignore
+import childTemplate from './resultTemplates/childTemplate.html';
+// @ts-ignore
+import parentTemplate from './resultTemplates/parentTemplate.html';
+// @ts-ignore
+import youtubeTemplate from './resultTemplates/youtubeResultTemplate.html';
+
+export default class ExampleRelevantGenerativeAnswering extends LightningElement {
+  /** @type {string} */
+  @api engineId = 'example-relevant-generative-answering';
+  /** @type {string} */
+  @api searchHub = 'default';
+  /** @type {string} */
+  @api pipeline = 'genqatest';
+  /** @type {boolean} */
+  @api disableStateInUrl = false;
+  /** @type {boolean} */
+  @api skipFirstSearch = false;
+  /** @type {string} */
+  @api answerStyle = 'default';
+  /** @type {boolean} */
+  @api multilineFooter = false;
+
+  handleResultTemplateRegistration(event) {
+    event.stopPropagation();
+
+    const resultTemplatesManager = event.detail;
+
+    const isCase = CoveoHeadless.ResultTemplatesHelpers.fieldMustMatch(
+      'objecttype',
+      ['Case']
+    );
+    const isYouTube = CoveoHeadless.ResultTemplatesHelpers.fieldMustMatch(
+      'filetype',
+      ['YouTubeVideo']
+    );
+    const isChatter = CoveoHeadless.ResultTemplatesHelpers.fieldMustMatch(
+      'objecttype',
+      ['FeedItem']
+    );
+    const isThread = CoveoHeadless.ResultTemplatesHelpers.fieldMustMatch(
+      'source',
+      ['iNaturalistTaxons']
+    );
+    const isChild = CoveoHeadless.ResultTemplatesHelpers.fieldMustMatch(
+      'quantic__templateId',
+      ['myChildTemplate']
+    );
+
+    resultTemplatesManager.registerTemplates(
```

Comment:
I added the results templates here also instead of importing them from `exampleSearch` since if something ever changes with the example search, at least it will not break this example. (there is code duplication as a result) 

If this is not something we want, I can change the import or refactor the `exampleSearch` to add properties and render in the HTML of the RGA example the `exampleSearch` with extra properties to make it "RGA". 

I am not a fan of this option however, I believe each example should be independant.

Reply:
In this example component, the goal is to show case how the generated answer component can be used, so I think, for simplicity sake and to keep the focus on the generated answer component, we could completely avoid adding this result templates logic. The default templates will be then be used and I think this should be fine.

Reply:
sounds good! will remove the logic for templates